src/lua/loaders.lua

```diff
@@ -0,0 +1,166 @@
+local fio = require("fio")
+
+local soext = (jit.os == "OSX" and "dylib" or "so")
```

Comment:
I don't think that I should enhance this code at spliting into separate files. Moreover, any way to obtain `'dylib'` or `'so'` string is likely wrong. AFAIR, we should search as for `.dylib` as well as `.so` on Mac OS, because some modules are compiled into an `.so` file on Mac OS due to a PUC-Rio Lua 5.1.0 problem (it was fixed in later versions, but I guess that some modules are still shipped as `.so`):

> For consistency with other platforms the modules should be installed with the `.so` extension on Mac OS X, too, and this assumed by the default definitions of the package cpath.

http://lua-users.org/wiki/BuildingModules

So if we'll change it, we'll change it more dramatically :)

There are more things we likely should change around the loaders. We aware of them, but I hope we'll do it later in scope of its own tasks.

Reply:
OK, I agree with you now. IIRC, so/dylib problem is in our TODO list for Application Server, so let's solve it there. Resolving this comment.

---

> > For consistency with other platforms the modules should be installed with the `.so` extension on Mac OS X, too, and this assumed by the default definitions of the package cpath.

What a bullshit.

Reply:
I re-verified, it [is](https://www.notion.so/tarantool/so-dylib-package-cpath-contains-only-dylib-on-mac-1a828cc9b4d6490db86e37a91e489572) in the list.